src/runtime/swg-client-event-manager.js

```diff
@@ -0,0 +1,312 @@
+/**
+ * Copyright 2019 The Subscribe with Google Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {AnalyticsEvent,EventOriginator} from '../proto/api_messages';
+
+/**
+ * The callback used to notify registered listeners about new client events.
+ * @callback SwgEventManagerListener
```

Comment:
can we use a definition like we do in rest of the code for a callback

{function(!SwgClientEvent)} SwgEventListener

Also do we need to call it SwgEventManagerListener? SwgEventListener not enough?

Reply:
That didn't work in that the exported variable was marked as "any" when I hovered over it in visual studio.  I modified this to use that syntax though without the labeling.  (its now directly in the function signatures and applied to the new const arrays.